idx/memory/tag_query.go

```diff
@@ -567,32 +322,30 @@ func (q *TagQuery) testByAllExpressions(id schema.MKey, def *idx.Archive, omitTa
 
 // testByMatch filters a given metric by matching a regular expression against
 // the values of specific associated tags
-func (q *TagQuery) testByMatch(def *idx.Archive, exprs []kvRe, not bool) bool {
+func (q *TagQueryContext) testByMatch(def *idx.Archive, exprs []kvRe, not bool) bool {
 EXPRS:
 	for _, e := range exprs {
-		if e.key == "name" {
-			if e.value == nil || e.value.MatchString(def.Name) {
+		if e.Key == "name" {
+			if e.Regex == nil || e.Regex.MatchString(def.Name) {
 				if not {
 					return false
-				} else {
-					continue EXPRS
 				}
+				continue EXPRS
```

Comment:
why are we changing the logic here and below?

Reply:
ah, i guess because it's equivalent anyway because the clause above returns

Reply:
yeah, i think `go-vet` or one of those utilities keeps highlighting it when there's some pattern like `if xyz { return } else { something }`, because the `else` can just be removed in that case